azurecompute-arm/src/main/java/org/jclouds/azurecompute/arm/domain/VirtualMachineScaleSetOSProfile.java

```diff
@@ -0,0 +1,280 @@
+             */
+            public abstract Protocol protocol();
+
+            /**
+             * The certificate url or the protcol listener
+             */
+            @Nullable
+            public abstract String certificateUrl();
+
+            @SerializedNames({"protocol", "certificateUrl"})
+            public static ProtocolListener create(final Protocol protocol, final String certificateUrl) {
+
+               return new AutoValue_VirtualMachineScaleSetOSProfile_WindowsConfiguration_WinRM_ProtocolListener(
+                  protocol, certificateUrl);
+            }
+         }
+
+         /**
+          * Map of different settings
+          */
+         public abstract List<ProtocolListener> listeners();
+
+         @SerializedNames({"listeners"})
+         public static WinRM create(final List<ProtocolListener> listeners) {
+            return new AutoValue_VirtualMachineScaleSetOSProfile_WindowsConfiguration_WinRM(listeners == null ? ImmutableList.<ProtocolListener>of() : ImmutableList.copyOf(listeners));
+         }
+      }
+
+      @AutoValue
+      public abstract static class AdditionalUnattendContent {
+
+         public abstract String pass();
+
+         public abstract String component();
+
+         public abstract String settingName();
+
+         public abstract String content();
+
+         @SerializedNames({"pass", "component", "settingName", "content"})
+         public static AdditionalUnattendContent create(final String pass, final String component,
+                                                        final String settingName,
+                                                        final String content) {
+
+            return new AutoValue_VirtualMachineScaleSetOSProfile_WindowsConfiguration_AdditionalUnattendContent(
+               pass, component, settingName, content);
+         }
+      }
+
+      /**
+       * The provision VM Agent true of false.
+       */
+      public abstract boolean provisionVMAgent();
+
+      /**
+       * winR
+       */
+      @Nullable
+      public abstract WinRM winRM();
+
+      /**
+       * unattend content
+       */
+      @Nullable
+      public abstract List<AdditionalUnattendContent> additionalUnattendContent();
+
+      /**
+       * is automatic updates enabled
+       */
+      public abstract boolean enableAutomaticUpdates();
+
+      @SerializedNames({"provisionVMAgent", "winRM", "additionalUnattendContent", "enableAutomaticUpdates"})
+      public static WindowsConfiguration create(final boolean provisionVMAgent, final WinRM winRM,
+                                                final List<AdditionalUnattendContent> additionalUnattendContent,
+                                                final boolean enableAutomaticUpdates) {
+
+         return new AutoValue_VirtualMachineScaleSetOSProfile_WindowsConfiguration(provisionVMAgent, winRM,
+            additionalUnattendContent, enableAutomaticUpdates);
+      }
+   }
```

Comment:
Can't these configuration classes be extracted and reused? They look quite similar to the existing ones.